src/components/TransactionHistoryRows/components/AssetTransfer.tsx

```diff
@@ -0,0 +1,53 @@
+import { Box, Stack } from '@chakra-ui/react'
+import { useMemo } from 'react'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import type { Transfer } from 'hooks/useTxDetails/useTxDetails'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { fromBaseUnit } from 'lib/math'
+
+import { FALLBACK_PRECISION, FALLBACK_SYMBOL } from '../constants'
+
+type AssetTransferProps = {
+  compactMode?: boolean
+  transfer: Transfer
+  index: number
+}
+
+export const AssetTransfer: React.FC<AssetTransferProps> = ({ index, compactMode, transfer }) => {
+  const cryptoAmount = useMemo(() => {
+    const precision = transfer.asset?.precision ?? FALLBACK_PRECISION
+    return fromBaseUnit(transfer.value, precision)
+  }, [transfer.asset?.precision, transfer.value])
+  const fiatAmount = useMemo(
+    () => bnOrZero(cryptoAmount).times(transfer.marketData.price).toString(),
+    [cryptoAmount, transfer.marketData.price],
+  )
+  return (
+    <Stack
+      alignItems='center'
+      key={index}
```

Comment:
We only use `index` as a key, we should be able to ditch it in favor of an actually *sane* and *unique` key